docs/reference/variables.md

```diff
@@ -112,11 +146,11 @@ always passed. It acts as a form of documentation for both the developer and the
 translator: it tells the developer that the message expects a `count` variable,
 and it tells the translator that the message gets a `count` variable passed in.
 
-Ideally expressions in `.input` should also specify a {function} like `:string`,
-`:number`, or `:datetime`. These functions are not type assertions. They act as
-type coercions, converting the external input value to the specified type, by
-passing the value through the specified function (creating a new {resolved
-value}). This also means that formatting options can be specified in the
+Ideally expressions in `.input` should also specify a [function](/docs/reference/functions).
+like `:string`, `:number`, or `:datetime`. These functions are not type assertions.
+They act as type coercions, returning a new value of the specified type
+based on the operand (or signaling an error if no coercion is possible).
+This also means that formatting options can be specified in the
 function.
```

Comment:
For a follow up as this PR is already quite large: we should mention that it can also specify selector options that are carried with the value to any `.match` statement that uses this variable.